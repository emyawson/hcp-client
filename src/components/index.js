export * from './accordion';
export * from './arrow-button';
export * from './banner';
export * from './badge';
export * from './block';
export * from './button'; // eslint-disable-line
export * from './card';
export * from './caret-button';
export * from './checkbox';
export * from './comment';
export * from './dropdown';
export * from './fonts';
export * from './graph-statistics';
export * from './graph-detail';
export * from './graph-legend';
export * from './graph-options';
export * from './loading-indicator';
export * from './loading-message';
export * from './loading-ring';
export * from './localized-text';
export * from './keyline';
export * from './meal-times';
export * from './menu';
export * from './modal';
export * from './notification-banner';
export * from './parentheses';
export * from './placeholder-icon';
export * from './popover';
export * from './section-header';
export * from './tabs';
export * from './graph-settings-row';
export * from './table';
export * from './table-header';
export * from './table-row';
export * from './table-body';
export { TableCell } from './table-cell';
export * from './row'; // eslint-disable-line
export * from './column';
export * from './input';
export * from './widget';
export * from './link';
export * from './indicator-label';
export * from './vertical-tabs';
export * from './number-input';
export * from './econecta';
export * from './grid-layout';
export * from './with-error';
export * from './with-loader';
export * from './progress-bar';
export * from './traffic-light-circle';
export * from './toggle-switch';
export * from './forms';
export * from './resize-wrapper';
export { ToolTip } from './tool-tip';
export * from './portal';
export * from './iframe';
export * from './back-button';
export * from './range-slider';
export * from './step-navigation';
