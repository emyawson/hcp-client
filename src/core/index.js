export * from './env';
export * from './patient-date-range';
export * from './patient';
export * from './request';
export * from './styles';
export * from './authentication';
export * from './strip-delivery';
export * from './patient-search';
export * from './prescription';
export * from './patient-stock';
export * from './state';
export * from './org-stock';
export * from './alerts';
export * from './modal';
export * from './user';
export * from './permissions';
export * from './dtc';
export * from './help';
export * from './diagnostics';
export * from './device-assignment'; // eslint-disable-line
