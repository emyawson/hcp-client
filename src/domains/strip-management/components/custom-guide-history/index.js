export * from './custom-guide-history.component';
