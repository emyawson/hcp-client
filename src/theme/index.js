import {
  boxShadows,
  boxShadow,
  fontSize,
  fontSizesMap,
  fontWeights,
  borderRadiusMap,
  breakpointsMap,
  spacingMap,
  spacing,
  colors,
  complexStyles,
  BASE_FONT_SIZE,
  transitions,
  transitionEasing,
  transitionSpeed,
  zIndexes,
  zIndexScale,
} from 'src/core/styles';

import { animations } from './animations';
import { letterSpacing } from './typography';

export const theme = {
  BASE_FONT_SIZE,
  animations,
  boxShadow,
  boxShadows,
  borderRadius: borderRadiusMap,
  breakpoints: breakpointsMap,
  colors,
  fontSize,
  fontSizes: fontSizesMap,
  fontWeights,
  letterSpacing,
  spacing,
  space: spacingMap,
  textStyles: complexStyles.textStyles,
  transitions,
  transitionEasing,
  transitionSpeed,
  zIndexScale,
  zIndexes,
};

export type ThemeInterface = typeof theme;
