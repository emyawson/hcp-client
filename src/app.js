import * as React from 'react';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';
import { ConnectedRouter } from 'react-router-redux';
import { Route } from 'react-router-dom';
import { PersistGate } from 'redux-persist/lib/integration/react';
import { injectGlobal, ThemeProvider } from 'styled-components';
import 'normalize.css';
import 'react-virtualized/styles.css';
import 'react-vis/dist/style.css';
import 'react-select/dist/react-select.css';
import 'core-js/fn/object/keys';
import 'core-js/fn/array/find';
import 'core-js/fn/array/find-index';
import 'core-js/fn/array/from';
import 'core-js/fn/array/includes';
import 'core-js/fn/string/ends-with';
import 'core-js/fn/string/starts-with';
import 'core-js/fn/string/includes';
import 'core-js/fn/string/repeat';
import 'core-js/fn/array/find-index';
import 'core-js/es6/math';
import 'core-js/fn/number';
import 'intl';
import 'core-js/es6/set';
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';

import { ModalContainer as Modal } from 'src/widgets';
import { theme } from 'src/theme';

import { Block } from './components/block';
import { Column } from './components/column';
import { LoadingRing } from './components/loading-ring';
import { Bundle } from './navigation/bundle';
import { configureStore } from './store';
import { i18n } from './i18n';
import { colors, BASE_FONT_SIZE, APP_MIN_WIDTH } from './core/styles';
import { injectScript } from './inject-script';
import { Config } from './core/env';

export const { persistor, store, history } = configureStore({});
const { REACT_APP_GIGYA_TOKEN } = Config;

// tslint:disable:no-unused-expression
injectGlobal`
  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }

  html {
    font-size: ${BASE_FONT_SIZE}px;
  }

  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    background-color: ${colors.silverLight};
    min-width: ${APP_MIN_WIDTH};
  }

  body,
  button,
  input,
  optgroup,
  select,
  textarea {
    font-family: "Nunito", -apple-system,BlinkMacSystemFont, "Segoe UI", sans-serif;
  }
  ::selection {
    background: ${colors.brandBlue};
    color: ${colors.white};
  }

  @media print {
    body {
      -webkit-print-color-adjust: exact;
      color-adjust: exact;
    }

    section,
    img,
    svg,
    table,
    ul  {
      page-break-inside: avoid;
    }

    .print-hide {
      display: none;
    }

    .print-no-m {
      margin: 0;
    }

     @page {
        size: portrait;
    }
  }`; // tslint:enable:no-unused-expression

class AppContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isError: false,
      isLoading: true,
    };
  }

  componentDidMount() {
    injectScript({
      onError: () => {
        this.setState({
          isError: false,
        });
      },
      onLoad: () => {
        this.setState({
          isLoading: false,
        });
      },
      path: 'https://cdns.gigya.com/js/gigya.js',
      queryParams: {
        apiKey: REACT_APP_GIGYA_TOKEN,
        lang: i18n.language,
      },
    });
  }

  render() {
    const { NavigationContainer } = this.props;
    const { isLoading, isError } = this.state;
    if (isError) {
      return <div>Something went wrong</div>;
    }
    if (isLoading) {
      return (
        <Column align="center" height="100vh" justifyContent="center">
          <LoadingRing infinite />
        </Column>
      );
    }
    return (
      <section id="AppContainer">
        <Route path="/" component={NavigationContainer} />
        <Modal />
        <Block id="version" display="none">
          {process.env.REACT_APP_VERSION}
        </Block>
      </section>
    );
  }
}

AppContainer.displayName = 'AppContainer';

export const App = ({ NavigationContainer }) => (
  <Provider store={store}>
    <PersistGate persistor={persistor}>
      <I18nextProvider i18n={i18n}>
        <React.Fragment>
          <ThemeProvider theme={theme}>
            <ConnectedRouter history={history}>
              <AppContainer NavigationContainer={NavigationContainer} />
            </ConnectedRouter>
          </ThemeProvider>
          <div id="tool-tip-root" />
        </React.Fragment>
      </I18nextProvider>
    </PersistGate>
  </Provider>
);

export const AppBundle = props => (
  <Bundle
    bundleWillLoad={async () => {
      const {
        NavigationContainer,
      } = await import(/* webpackChunkName: "app-bundle" */ 'src/navigation/navigation.container');
      return NavigationContainer;
    }}
    bundleDidLoad={NavigationContainer => (
      <App NavigationContainer={NavigationContainer} />
    )}
  />
);
