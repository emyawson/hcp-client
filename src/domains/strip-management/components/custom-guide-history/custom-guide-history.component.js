import React from 'react';
import { equals, last, length } from 'ramda';

import {
  Accordion,
  AccordionItem,
  CaretIconLeft,
  CaretIconRight,
} from 'src/components';
import { TrashIcon } from 'src/assets/icons';
import { fontSize, colors } from 'src/core/styles';
import { translate } from 'src/i18n';
import { formatDateString } from 'src/utils';
import { RenderIf } from 'src/utils/render-if';

import {
  CustomGuideHistoryWrapperDiv,
  CustomGuideDetailsItemWrapper,
  RemoveCustomGuideRowDiv,
  RemoveCustomGuideButton,
  GuideHistoryPageNavigationWrapperDiv,
  GuideHistoryPageNumberButton,
  GuideHistoryPageNavButton,
  GuideHistoryPageNavLabel,
} from './custom-guide-history.style';

import {
  HistoryHeader,
  CustomClinicGuideSummary,
  LabelledGridItem,
} from '../../components';

const areMultiplePages = pages => length(pages) > 1;

export const CustomGuideHistory = ({
  customClinicGuides,
  guideHistoryFilter,
  setGuideHistoryFilter,
  onClickRemoveCustomGuide,
  guidePageNumbers,
  currentGuidePage,
  setGuideHistoryPage,
}) => (
  <CustomGuideHistoryWrapperDiv>
    <HistoryHeader
      guideHistoryFilter={guideHistoryFilter}
      setGuideHistoryFilter={setGuideHistoryFilter}
    >
      {translate('prescription.customClinicGuides.historyHeader')}
    </HistoryHeader>
    <Accordion activeItems={['example1']} allowMultiple>
      {customClinicGuides.map(
        ({
          name,
          minimumStrips,
          maximumStrips,
          period,
          id,
          ownerName,
          updatedAt,
          isRemovable,
          therapyName,
        }) => (
          <AccordionItem
            id={id}
            title={
              <CustomClinicGuideSummary
                name={name}
                ownerName={ownerName}
                date={formatDateString({ dateString: updatedAt })}
              />
            }
            cardStyles={['flat', 'noPadding']}
            key={id}
          >
            <CustomGuideDetailsItemWrapper>
              <LabelledGridItem
                label={translate(
                  'prescription.customClinicGuides.patientTherapy',
                )}
                value={therapyName}
                stretchToFitRow={false}
              />
              <LabelledGridItem
                label={translate(
                  'prescription.customClinicGuides.quantityRange',
                )}
                value={`${minimumStrips}-${maximumStrips}`}
                stretchToFitRow={false}
              />
              <LabelledGridItem
                label={translate('prescription.customClinicGuides.period')}
                value={period}
                stretchToFitRow={false}
              />
            </CustomGuideDetailsItemWrapper>
            <RemoveCustomGuideRowDiv>
              <RemoveCustomGuideButton
                buttonStyle="outlinedLight"
                label="remove"
                icon={<TrashIcon />}
                fontSize={fontSize.label}
                uppercase
                disabled={!isRemovable}
                onClick={() =>
                  onClickRemoveCustomGuide({
                    clinicGuideName: name,
                    clinicGuideId: id,
                  })
                }
              />
            </RemoveCustomGuideRowDiv>
          </AccordionItem>
        ),
      )}
    </Accordion>
    <RenderIf validate={areMultiplePages(guidePageNumbers)}>
      <GuideHistoryPageNavigationWrapperDiv>
        <GuideHistoryPageNavButton
          onClick={() => setGuideHistoryPage({ page: currentGuidePage - 1 })}
          disabled={equals(currentGuidePage, 1)}
        >
          <CaretIconLeft fillColor={colors.brandBlue} />
          <GuideHistoryPageNavLabel>
            {translate('prescription.customClinicGuides.previous')}
          </GuideHistoryPageNavLabel>
        </GuideHistoryPageNavButton>
        {guidePageNumbers.map(number => (
          <GuideHistoryPageNumberButton
            label={number}
            isSelected={equals(number, currentGuidePage)}
            onClick={() => setGuideHistoryPage({ page: number })}
            key={number}
          />
        ))}
        <GuideHistoryPageNavButton
          onClick={() => setGuideHistoryPage({ page: currentGuidePage + 1 })}
          disabled={equals(currentGuidePage, last(guidePageNumbers))}
        >
          <GuideHistoryPageNavLabel>
            {translate('prescription.customClinicGuides.next')}
          </GuideHistoryPageNavLabel>
          <CaretIconRight fillColor={colors.brandBlue} />
        </GuideHistoryPageNavButton>
      </GuideHistoryPageNavigationWrapperDiv>
    </RenderIf>
  </CustomGuideHistoryWrapperDiv>
);
