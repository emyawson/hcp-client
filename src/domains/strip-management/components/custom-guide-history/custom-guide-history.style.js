import styled from 'styled-components';

import { borderRadius, spacing, colors, fontSize } from 'src/core/styles';
import { Button, weight } from 'src/components';
import { combineRems } from 'src/utils';

export const CustomGuideHistoryWrapperDiv = styled.div`
  padding: ${props => props.theme.spacing.four} 0;
`;

export const CustomGuideDetailsItemWrapper = styled.div`
  display: flex;
  padding: ${spacing.three} 0 ${spacing.four} 0;
  width: 100%;
`;

export const RemoveCustomGuideRowDiv = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;
  border-top: 1px solid ${props => props.theme.colors.silverMedium};
`;

export const RemoveCustomGuideButton = styled(Button)`
  border-color: ${colors.lavender} ${colors.lavender} ${colors.lavender}
    ${colors.silverMedium};
  border-width: 1px;
  border-radius: 0 0 ${borderRadius.six} 0;
  background: ${colors.lavender};
  padding: ${combineRems(spacing.two, spacing.three)} 0;
  &:hover {
    border: 1px solid ${colors.brandBlue};
    background: ${colors.lavender};
  }
  &[disabled] {
    border-color: ${colors.white} ${colors.white} ${colors.white}
      ${colors.silver};
  }
`;

export const GuideHistoryPageNavigationWrapperDiv = styled.div`
  width: 100%;
  display: flex;
  border: 1px solid ${colors.silver};
  justify-content: center;
  border-radius: ${borderRadius.two};
  padding: ${combineRems(spacing.one, spacing.three)};
`;

export const GuideHistoryPageNumberButton = styled(Button)`
  border-radius: 50%;
  height: ${combineRems(spacing.one, spacing.four)};
  width: ${combineRems(spacing.one, spacing.four)};
  padding: ${spacing.one};
  margin: 0 ${spacing.two};
  color: ${props => (props.isSelected ? colors.silver : colors.brandBlue)};
  background-color: ${props =>
    props.isSelected ? colors.brandBlue : colors.silver};
  min-width: auto;
  border: 0;
`;

export const GuideHistoryPageNavButton = styled.button`
  border: none;
  background-color: transparent;
  color: ${colors.brandBlue};
  cursor: pointer;
  font-weight: ${weight.bold};
  font-size: ${fontSize.p};
  &:focus {
    outline: none;
    border: 1px solid ${colors.brandBlue};
  }
  &:hover {
    color: ${colors.brandBlueDark};
  }
  &:hover path {
    fill: ${colors.brandBlueDark};
  }
  &[disabled] {
    opacity: 0.3;
    pointer-events: none;
  }
`;

export const GuideHistoryPageNavLabel = styled.span`
  padding: ${spacing.two};
`;
