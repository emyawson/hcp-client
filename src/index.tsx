import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { AppBundle } from './app';

ReactDOM.render(<AppBundle />, document.getElementById('root'));
