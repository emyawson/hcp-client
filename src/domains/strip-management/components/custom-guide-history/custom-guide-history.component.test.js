import React from 'react';
import { shallow } from 'enzyme';

import { mountWithTheme } from 'src/test/render-with-theme';
import { theme } from 'src/theme';

import { CustomGuideHistory } from './custom-guide-history.component';
import { GuideHistoryPageNavigationWrapperDiv } from './custom-guide-history.style';

describe('Custom guide history component', () => {
  const baseProps = {
    customClinicGuides: [],
    guideHistoryFilter: 'ALL',
    setGuideHistoryFilter: () => {},
    onClickRemoveCustomGuide: () => {},
    currentGuidePage: 1,
    setGuideHistoryPage: () => {},
  };
  it('Renders correctly', () => {
    const mockProps = {
      ...baseProps,
      guidePageNumbers: [1, 2, 3],
    };
    const wrapper = shallow(<CustomGuideHistory {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Hides page navigation when there are less than 2 pages', () => {
    const mockProps = {
      ...baseProps,
      guidePageNumbers: [1],
    };
    const wrapper = mountWithTheme(
      <CustomGuideHistory {...mockProps} />,
      theme,
    );
    expect(wrapper.find(GuideHistoryPageNavigationWrapperDiv)).toHaveLength(0);
  });
  it('Show page navigation when there are 2 pages or more', () => {
    const mockProps = {
      ...baseProps,
      guidePageNumbers: [1, 2],
    };
    const wrapper = mountWithTheme(
      <CustomGuideHistory {...mockProps} />,
      theme,
    );
    expect(wrapper.find(GuideHistoryPageNavigationWrapperDiv)).toHaveLength(1);
  });
});
