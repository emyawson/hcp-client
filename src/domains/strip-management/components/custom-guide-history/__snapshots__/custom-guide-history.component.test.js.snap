// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom guide history component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentGuidePage={1}
    customClinicGuides={Array []}
    guideHistoryFilter="ALL"
    guidePageNumbers={
      Array [
        1,
        2,
        3,
      ]
    }
    onClickRemoveCustomGuide={[Function]}
    setGuideHistoryFilter={[Function]}
    setGuideHistoryPage={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          guideHistoryFilter="ALL"
          setGuideHistoryFilter={[Function]}
        >
          Custom Clinic Guide History
        </Unknown>,
        <Accordion
          activeItems={
            Array [
              "example1",
            ]
          }
          allowAllItemsClosed={false}
          allowMultiple={true}
          label=""
          labelDisplayLogic={[Function]}
          labelled={false}
          onChange={[Function]}
          sendInitialChangeEvent={true}
        />,
        <f1
          validate={true}
        >
          <styled.div>
            <styled.button
              disabled={true}
              onClick={[Function]}
            >
              <Unknown
                fillColor="#0066CC"
              />
              <styled.span>
                Prev
              </styled.span>
            </styled.button>
            <Styled(Button)
              isSelected={true}
              label={1}
              onClick={[Function]}
            />
            <Styled(Button)
              isSelected={false}
              label={2}
              onClick={[Function]}
            />
            <Styled(Button)
              isSelected={false}
              label={3}
              onClick={[Function]}
            />
            <styled.button
              disabled={false}
              onClick={[Function]}
            >
              <styled.span>
                Next
              </styled.span>
              <Unknown
                fillColor="#0066CC"
              />
            </styled.button>
          </styled.div>
        </f1>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": "Custom Clinic Guide History",
          "guideHistoryFilter": "ALL",
          "setGuideHistoryFilter": [Function],
        },
        "ref": null,
        "rendered": "Custom Clinic Guide History",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "activeItems": Array [
            "example1",
          ],
          "allowAllItemsClosed": false,
          "allowMultiple": true,
          "children": Array [],
          "label": "",
          "labelDisplayLogic": [Function],
          "labelled": false,
          "onChange": [Function],
          "sendInitialChangeEvent": true,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.div>
            <styled.button
              disabled={true}
              onClick={[Function]}
            >
              <Unknown
                fillColor="#0066CC"
              />
              <styled.span>
                Prev
              </styled.span>
            </styled.button>
            <Styled(Button)
              isSelected={true}
              label={1}
              onClick={[Function]}
            />
            <Styled(Button)
              isSelected={false}
              label={2}
              onClick={[Function]}
            />
            <Styled(Button)
              isSelected={false}
              label={3}
              onClick={[Function]}
            />
            <styled.button
              disabled={false}
              onClick={[Function]}
            >
              <styled.span>
                Next
              </styled.span>
              <Unknown
                fillColor="#0066CC"
              />
            </styled.button>
          </styled.div>,
          "validate": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.button
                disabled={true}
                onClick={[Function]}
              >
                <Unknown
                  fillColor="#0066CC"
                />
                <styled.span>
                  Prev
                </styled.span>
              </styled.button>,
              Array [
                <Styled(Button)
                  isSelected={true}
                  label={1}
                  onClick={[Function]}
                />,
                <Styled(Button)
                  isSelected={false}
                  label={2}
                  onClick={[Function]}
                />,
                <Styled(Button)
                  isSelected={false}
                  label={3}
                  onClick={[Function]}
                />,
              ],
              <styled.button
                disabled={false}
                onClick={[Function]}
              >
                <styled.span>
                  Next
                </styled.span>
                <Unknown
                  fillColor="#0066CC"
                />
              </styled.button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Unknown
                    fillColor="#0066CC"
                  />,
                  <styled.span>
                    Prev
                  </styled.span>,
                ],
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "fillColor": "#0066CC",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Prev",
                  },
                  "ref": null,
                  "rendered": "Prev",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "isSelected": true,
                "label": 1,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "isSelected": false,
                "label": 2,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "isSelected": false,
                "label": 3,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Next
                  </styled.span>,
                  <Unknown
                    fillColor="#0066CC"
                  />,
                ],
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Next",
                  },
                  "ref": null,
                  "rendered": "Next",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "fillColor": "#0066CC",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            guideHistoryFilter="ALL"
            setGuideHistoryFilter={[Function]}
          >
            Custom Clinic Guide History
          </Unknown>,
          <Accordion
            activeItems={
              Array [
                "example1",
              ]
            }
            allowAllItemsClosed={false}
            allowMultiple={true}
            label=""
            labelDisplayLogic={[Function]}
            labelled={false}
            onChange={[Function]}
            sendInitialChangeEvent={true}
          />,
          <f1
            validate={true}
          >
            <styled.div>
              <styled.button
                disabled={true}
                onClick={[Function]}
              >
                <Unknown
                  fillColor="#0066CC"
                />
                <styled.span>
                  Prev
                </styled.span>
              </styled.button>
              <Styled(Button)
                isSelected={true}
                label={1}
                onClick={[Function]}
              />
              <Styled(Button)
                isSelected={false}
                label={2}
                onClick={[Function]}
              />
              <Styled(Button)
                isSelected={false}
                label={3}
                onClick={[Function]}
              />
              <styled.button
                disabled={false}
                onClick={[Function]}
              >
                <styled.span>
                  Next
                </styled.span>
                <Unknown
                  fillColor="#0066CC"
                />
              </styled.button>
            </styled.div>
          </f1>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": "Custom Clinic Guide History",
            "guideHistoryFilter": "ALL",
            "setGuideHistoryFilter": [Function],
          },
          "ref": null,
          "rendered": "Custom Clinic Guide History",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "activeItems": Array [
              "example1",
            ],
            "allowAllItemsClosed": false,
            "allowMultiple": true,
            "children": Array [],
            "label": "",
            "labelDisplayLogic": [Function],
            "labelled": false,
            "onChange": [Function],
            "sendInitialChangeEvent": true,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.button
                disabled={true}
                onClick={[Function]}
              >
                <Unknown
                  fillColor="#0066CC"
                />
                <styled.span>
                  Prev
                </styled.span>
              </styled.button>
              <Styled(Button)
                isSelected={true}
                label={1}
                onClick={[Function]}
              />
              <Styled(Button)
                isSelected={false}
                label={2}
                onClick={[Function]}
              />
              <Styled(Button)
                isSelected={false}
                label={3}
                onClick={[Function]}
              />
              <styled.button
                disabled={false}
                onClick={[Function]}
              >
                <styled.span>
                  Next
                </styled.span>
                <Unknown
                  fillColor="#0066CC"
                />
              </styled.button>
            </styled.div>,
            "validate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.button
                  disabled={true}
                  onClick={[Function]}
                >
                  <Unknown
                    fillColor="#0066CC"
                  />
                  <styled.span>
                    Prev
                  </styled.span>
                </styled.button>,
                Array [
                  <Styled(Button)
                    isSelected={true}
                    label={1}
                    onClick={[Function]}
                  />,
                  <Styled(Button)
                    isSelected={false}
                    label={2}
                    onClick={[Function]}
                  />,
                  <Styled(Button)
                    isSelected={false}
                    label={3}
                    onClick={[Function]}
                  />,
                ],
                <styled.button
                  disabled={false}
                  onClick={[Function]}
                >
                  <styled.span>
                    Next
                  </styled.span>
                  <Unknown
                    fillColor="#0066CC"
                  />
                </styled.button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Unknown
                      fillColor="#0066CC"
                    />,
                    <styled.span>
                      Prev
                    </styled.span>,
                  ],
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "fillColor": "#0066CC",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Prev",
                    },
                    "ref": null,
                    "rendered": "Prev",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "isSelected": true,
                  "label": 1,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "isSelected": false,
                  "label": 2,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "isSelected": false,
                  "label": 3,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.span>
                      Next
                    </styled.span>,
                    <Unknown
                      fillColor="#0066CC"
                    />,
                  ],
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Next",
                    },
                    "ref": null,
                    "rendered": "Next",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "fillColor": "#0066CC",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
